lua/toggleterm/config.lua

```diff
@@ -17,10 +20,34 @@ local config = {
   shell = vim.o.shell,
   float_opts = {
     winblend = 0,
-    highlights = {
-      background = "Normal",
-      border = "Normal",
-    },
+  },
+}
+
+config.highlights = {
+  Normal = {
+    guibg = colors.shade_color(colors.get_hex("Normal", "bg"), constants.shading_amount),
+  },
+  NormalFloat = {
+    guibg = colors.get_hex("NormalFloat", "bg"),
+  },
+  FloatBorder = {
+    guifg = colors.get_hex("FloatBorder", "fg"),
+    guibg = colors.get_hex("FloatBorder", "bg"),
+  },
+  EndOfBuffer = {
+    guibg = colors.shade_color(colors.get_hex("Normal", "bg"), constants.shading_amount),
```

Comment:
It looks like a lot of these colour variables are repeated, so it would be more efficient to fetch them once at the top of the function and re-use them rather than calling `colors.get_hex` a bunch of times with the same arguments.
```lua
local normal_bg = colors.get_hex(BLAH)

-- ...
guibg = colors.shade_color(normal_bg)
```